crates/core/src/pipeline/runner.rs

```diff
@@ -108,9 +121,117 @@ impl PipelineRunner {
             fps_last_frame_ctr: 0,
             info: None,
             thumb_interval: 1800, // Disable thumbnails by default for performance
+            idle_mode: false,
+            idle_start_time: None,
+            idle_variant_index: 0,
+            last_idle_frame_time: None,
         })
     }
 
+    /// Process a single idle frame for the current variant (rotating through variants)
+    unsafe fn process_single_idle_frame(&mut self, config: &PipelineConfig) -> Result<()> {
+        use std::time::{Duration, Instant};
+        
+        if config.variants.is_empty() {
+            return Ok(());
+        }
+
+        // Time-based frame rate calculation
+        let now = Instant::now();
+        if let Some(last_time) = self.last_idle_frame_time {
+            // Calculate target frame interval (assume 30fps for now, can be refined per variant)
+            let target_interval = Duration::from_millis(33); // ~30fps
+            let elapsed = now.duration_since(last_time);
+            
+            if elapsed < target_interval {
+                // Not time for next frame yet
+                std::thread::sleep(target_interval - elapsed);
+            }
+        }
+        self.last_idle_frame_time = Some(Instant::now());
+
+        // Rotate through variants to generate one frame at a time
```

Comment:
This is wrong the relationship should be 1 source video frame into multiple scalar/encoder chains, as i said before we only need 1 placeholder frame and we can use the same frame in each variant, thats how it works if the demuxer/decoder outputs a single frame its shared will all variants